components/statistic/StatisticList.js

```diff
@@ -0,0 +1,99 @@
+import styled from "styled-components";
+import StatisticCircles from "./StatisticCircles";
+import categorizeIntensity from "@/lib/categorizeIntensity";
+
+export default function StatisticList({
+  emotions,
+  emotionTypes,
+  customDateRange,
+}) {
+  const filteredEmotions =
+    customDateRange.start && customDateRange.end
+      ? emotions.filter((emotion) => {
+          const emotionDate = new Date(emotion.dateTime);
+          return (
+            emotionDate >= customDateRange.start &&
+            emotionDate <= customDateRange.end
+          );
+        })
+      : emotions;
```

Comment:
you can move this logic into your statistic page before rendering this component. then you can pass filteredEmtions to `StatisticList`. Then you don't need to pass down the customDateRange:

```js
// pages/statistic/index.js

  const filteredEmotions =
    customDateRange.start && customDateRange.end
      ? emotions.filter((emotion) => {
          const emotionDate = new Date(emotion.dateTime);
          return (
            emotionDate >= customDateRange.start &&
            emotionDate <= customDateRange.end
          );
        })
      : emotions;

return ...
   <StatisticList emtions={filteredEmotions} emotionTypes={emotionTypes}/> // no customDateRange needed here

```

Reply:
Yes, of course! Good move. 👯